src/event/metric.rs

```diff
@@ -241,30 +240,37 @@ impl Display for Metric {
         if let Some(timestamp) = &self.timestamp {
             write!(fmt, "{:?} ", timestamp)?;
         }
-        write!(fmt, "{}", quote_word(&self.name))?;
+        write_word(fmt, &self.name)?;
+        write!(fmt, "{{")?;
         if let Some(tags) = &self.tags {
-            write!(fmt, "{{")?;
             write_list(fmt, ",", tags.iter(), |fmt, (tag, value)| {
-                write!(fmt, "{}={:?}", quote_word(tag), value)
+                write_word(fmt, tag).and_then(|()| write!(fmt, "={:?}", value))
             })?;
-            write!(fmt, "}}")?;
         }
+        write!(fmt, "}}")?;
         match self.kind {
             MetricKind::Absolute => write!(fmt, " = "),
             MetricKind::Incremental => write!(fmt, " + "),
         }?;
         match &self.value {
             MetricValue::Counter { value } => write!(fmt, "{}", value),
             MetricValue::Gauge { value } => write!(fmt, "{}", value),
-            MetricValue::Set { values } => write_list(fmt, " ", values.iter(), |fmt, value| {
-                write!(fmt, "{}", quote_word(value))
-            }),
+            MetricValue::Set { values } => {
+                write_list(fmt, " ", values.iter(), |fmt, value| write_word(fmt, value))
+            }
             MetricValue::Distribution {
                 values,
                 sample_rates,
                 statistic,
             } => {
-                write!(fmt, "{:?} ", statistic)?;
+                write!(
+                    fmt,
+                    "{}",
+                    match statistic {
+                        StatisticKind::Histogram => "histogram ",
```

Comment:
It will be easy to miss that there should be a space after the name if ever expanded `StatisticKind`, so it's better to move space up to be with `{}`

Reply:
Done